File: Calculo Numerico/Trabalho 09/Trabalho9.cpp
/*
* Trabalho 10.
* Implemente o Método dos Retângulos e dos Trapézios 
* para equações do formato:
* f(x) = a.sen(x) + b.cos(x) + c.x
*
* Calculo Numerico
* Claudio Schepke
* ---
* Ciencia da Computacao
* Universidade Federal do Pampa
* 
*
* Integrantes:
* ---
* Marcos V. Treviso, 121150107, marcosvtreviso@gmail.com
*
*
* Instrucoes:
* ---
* Compile: g++ -o trab Trabalho10.cpp -Wall -Wextra
* Run: ./trab
*
*/

#include <iostream>
#include <iomanip>
#include <cstdio>
#include <cstdlib>
#include <cmath>
#include <cstring>
#include <vector>

using namespace std;




const bool DEBUG = false;
double c1, c2, c3; //a, b, c


double f(double x){
	return c1*sin(x) + c2*cos(x) + c3*x;
}


double metodoRetangulos(int a, int b, int n){

	double h = (b-a)/(double) n;
	double hh = h/2;
	
	double soma=0, x;
	for(int i=a; i<b; i++){
		x = (a + i*h) + hh;
		soma += f(x)*h;
	}

	return soma;
}

double metodoTrapezios(int a, int b, int n){

	double h = (b-a)/(double) n;
	
	double soma=0, x;
	for(int i=a; i<b; i++){
		x = (f(i) + f(i+1))/2.0;
		soma += x*h;
	}

	return soma;
}



int main(){

	cout << fixed << setprecision(8);

	int a, b, n;
	while(true){
			
		cout << "Informe os valores dos coeficientes c1, c2, c3:" << endl;
		cin >> c1 >> c2 >> c3;

		cout << "Informe os intervalos de integracao a, b:" << endl;
		cin >> a >> b;

		cout << "Informe o numero de divisoes n:" << endl;
		cin >> n;

		cout << "Metodo dos Retangulos: " << 
		metodoRetangulos(a, b, n) << endl;

		cout << "Metodo dos Trapezios: " << 
		metodoTrapezios(a, b, n) << endl << endl;

	}

	return 0;
}

File: Calculo Numerico/Trabalho 09/Trabalho9.py
import math


a, b, c = 0, 0, 0

def eq(x):
	return a*math.sin(x) + b*math.cos(x) + c*x



def getH(i, j, n):
	return (j-i)/(n*1.0)


def metodo_retangulos(i, j, n):
	
	h = getH(i, j, n)
	h2 = h/2.0

	s = 0
	for k in range(i, j):
		x = i + k*h
		s += eq(x+h2)*h
	 
	return s


def metodo_trapezios(i, j, n):

	h = getH(i, j, n)
	
	s = 0
	for k in range(i, j):
		x = (eq(k) + eq(k+1)) / 2.0
		s += x*h
	 
	return s



while(True):
	
	print("Digite os valores dos coeficientes a, b, c:")
	a, b, c = map(int, input().split())

	print("Digite os intervalos de integracao i, j:")
	i, j = map(int, input().split())

	print("Digite o numero de divisoes n:")
	n = int(input())

	print("Metodo dos Retangulos : %.8f" % metodo_retangulos(i, j, n))
	print("Metodo dos Trapezios  : %.8f" % metodo_trapezios(i, j, n))
	print("")

